aya_tamimi/models/basemodels.py

```diff
@@ -0,0 +1,21 @@
+from pydantic import BaseModel
+from models.enums import PriorityEnum,StatusEnum
+from datetime import date
+
+
+# this class for creation needs only
+class Create_Task_Basemodel(BaseModel):
+    title: str
+    description: str
+    due_date: date
```

Comment:
here you can check if the data is in YYYY:MM: DD format.

Reply:
The 'date' type expects the date to be only in YYYY-MM-DD format, so it automatically handles the validation

Reply:
oh, I see

Reply:
@JbreenJboor1212 the first validation should be enough and always should be more options. 

Thank you

Reply:
Appreciate the insights! I'll keep it in mind for future work. 
Thanks for the collaboration!